### PR TITLE
Fix compilation of some mutually recursive functions

### PR DESCRIPTION
**Pull request by vouillon**
Implement caml_alloc_dummy_infix

### CODE REVIEW

**Comment by hhugo** [comment]
Shouldn't we change the bytecode to also emit the arity ? and perform the same logic we already use for `caml_alloc_dummy_function` ?

File: compiler/tests-jsoo/test_rec_fun.ml [hunk 1]
```diff
@@ -0,0 +1,98 @@
+let rec h =
```
Comment (hhugo) [inline compiler/tests-jsoo/test_rec_fun.ml:1]: Missing header

**Comment by vouillon** [comment]
> Shouldn't we change the bytecode to also emit the arity ? and perform the same logic we already use for `caml_alloc_dummy_function` ?

Ideally, yes. On the other hand, this does not seem to be widely used...

**Comment by hhugo** [comment]
> > Shouldn't we change the bytecode to also emit the arity ? and perform the same logic we already use for `caml_alloc_dummy_function` ?
> 
> Ideally, yes. On the other hand, this does not seem to be widely used...

Patch available in https://github.com/hhugo/ocaml/commit/881d767cd4fda0f3fc6d1855063e1abe1771643b

(x-link https://github.com/ocaml/ocaml/pull/37)

**Pull request merged**